internal/pipe/nfpm/nfpm.go

```diff
@@ -191,6 +192,30 @@ func create(ctx *context.Context, fpm config.NFPM, format string, artifacts []*a
 		}
 	}
 
+	// AIX is weird, so we default to 7.2 as the earliest release
+	// that supports golang. This can be overridden by setting platform
+	// in your .goreleaser.yaml. See the following:
+	// https://www.unix.com/aix/266963-tip-problem-rpm-different-operating-system.html
+	// Additionally, it is recommended to set the rpmArch to ppc
+	// As AIX, while being ppc64, expects the rpms to specify ppc.
+	// We will default to setting ppc here, but again this can be
+	// overridden by setting it in your .goreleaser.yaml See the following:
+	// https://developer.ibm.com/articles/au-aix-build-open-source-rpm-packages/
+	// https://developer.ibm.com/articles/configure-yum-on-aix/
+	if infoPlatform == "aix" {
+		if artifacts[0].Goarch != "ppc64" {
+			log.Debugf("skipping aix/%s as its not supported", infoArch)
+			return nil
+		}
+		if format == "rpm" {
+			infoPlatform = "aix7.2"
+			infoArch = "ppc"
```

Comment:
Copying from my comment in the other thread:

Okay, that makes sense. However, you're setting infoArch to ppc, which also doesn't work in my mind. I separated infoArch and RPMarch on purpose. My understanding is that infoArch needs to remain ppc64 while the rpmArch needs to be ppc.

Reply:
Specifically, this is what it takes in raw NFPM to work:
```
name: "aix-somefile"
arch: "ppc64"
platform: "aix7.2"
version: "1.0.0"
vendor: Company
maintainer: Company Maintainer Email
description: Package Description
homepage: https://some-url.com
rpm:
  arch: ppc
```

Reply:
You'll notice Platform is also aix7.2, but that can be 7.3 also. Which is why I made it settable in my initial pass.

Reply:
it doesn't, its only used to package this one package.

the usage of rpm.arch is more like "i use this same config for every format, but on rpm i want arch to be different", which is not our case

Reply:
Does the arch not need to match the actual binary arch?

What about situations where we're doing multiple OSes with the ppc64 arch, and it only needs to be swapped to ppc for AIX?

Reply:
> Does the arch not need to match the actual binary arch?

no, the arch set into `nfpm.Info` is used only in the package... if it doesn't match (i.e. if its wrong), you'll get a package that installs but a binary that doesn't execute.

but this is handled here, we only change to `aix/ppc64` to `aix72/ppc` when format is `rpm`. 

> What about situations where we're doing multiple OSes with the ppc64 arch, and it only needs to be swapped to ppc for AIX?

its handled, we create a new `nfpm.Info` for each os/arch we package to... so it should all be fine.

Reply:
Okay. Thanks for explaining. I was under the impression the top level arch needed to match the binary, and the rpm level matches whatever weirdness for package names.

Reply:
One more question, what if they set their platform to aix7.3, and arch to ppc64. The arch will still need swapped to ppc. That doesn't appear to be done currently.

Reply:
there's no way to do that inside goreleaser...

the platform for aix binaries will always be `aix`, because that's what the build gives it (the `GOOS` basically)

if we eventually want to allow to customize the platform, that's another problem, but right now its not possible, so this is a non-issue